sdk/communication/Azure.Communication.Calling.Server/src/PartitionedDownloader.cs

```diff
@@ -0,0 +1,329 @@
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Communication.Calling.Server.Models;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Calling.Server
+{
+    internal class PartitionedDownloader
+    {
+        /// <summary>
+        /// The client used to download the blob.
+        /// </summary>
+        private readonly ConversationClient _client;
+
+        /// <summary>
+        /// The maximum number of simultaneous workers.
+        /// </summary>
+        private readonly int _maxWorkerCount;
+
+        /// <summary>
+        /// The size of the first range requested (which can be larger than the
+        /// other ranges).
+        /// </summary>
+        private readonly long _initialRangeSize;
+
+        /// <summary>
+        /// The size of subsequent ranges.
+        /// </summary>
+        private readonly long _rangeSize;
+
+        internal PartitionedDownloader(
+            ConversationClient client,
+            ContentTransferOptions transferOptions = default)
+        {
+            _client = client;
+
+            // Set _maxWorkerCount
+            if (transferOptions.MaximumConcurrency.HasValue
+                && transferOptions.MaximumConcurrency > 0)
+            {
+                _maxWorkerCount = transferOptions.MaximumConcurrency.Value;
+            }
+            else
+            {
+                _maxWorkerCount = Constants.ContentDownloader.Partition.DefaultConcurrentTransfersCount;
+            }
+
+            // Set _rangeSize
+            if (transferOptions.MaximumTransferSize.HasValue
+                && transferOptions.MaximumTransferSize.Value > 0)
+            {
+                _rangeSize = Math.Min(transferOptions.MaximumTransferSize.Value, Constants.ContentDownloader.Partition.MaxDownloadBytes);
+            }
+            else
+            {
+                _rangeSize = Constants.ContentDownloader.Partition.DefaultBufferSize;
+            }
+
+            // Set _initialRangeSize
+            if (transferOptions.InitialTransferSize.HasValue
+                && transferOptions.InitialTransferSize.Value > 0)
+            {
+                _initialRangeSize = transferOptions.InitialTransferSize.Value;
+            }
+            else
+            {
+                _initialRangeSize = Constants.ContentDownloader.Partition.DefaultInitalDownloadRangeSize;
+            }
+        }
+
+        internal async Task<Response> DownloadToAsync(
+            Stream destination,
+            Uri endpoint,
+            HttpRange initialBlock,
+            CancellationToken cancellationToken)
+        {
+            // Wrap the download range calls in a Download span for distributed
+            // tracing
+            DiagnosticScope scope = _client._clientDiagnostics.CreateScope($"{nameof(ConversationClient)}.{nameof(ConversationClient.DownloadTo)}");
```

Comment:
```suggestion
            DiagnosticScope scope = _client._clientDiagnostics.CreateScope($"{nameof(PartitionDownloader)}.{nameof(ConversationClient.DownloadTo)}");
```

Reply:
This is using ConversationClient method name (which is the caller). Is that a bad practice?